ui/src/parts/FeedbackButton/FeedbackButton.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import Button, { ButtonProps } from "react-bootstrap/Button";
+import Spinner from "react-bootstrap/Spinner";
+import Form from "react-bootstrap/Form";
+import { IconType } from "react-icons";
+import { BsExclamationCircle, BsCheck } from "react-icons/bs";
+import { PromiseResultShape } from "react-use-promise-matcher";
+import useFormikValuesChanged from "./useFormikValuesChanged";
+
+interface FeedbackButtonProps extends ButtonProps {
+  label: string;
+  Icon: IconType;
+  result: PromiseResultShape<any, any>;
+  clear: () => void;
+  successLabel?: string;
+}
+
+const FeedbackButton: React.FC<FeedbackButtonProps> = ({
+  result,
+  clear,
+  label,
+  Icon,
+  successLabel = "Success",
+  ...buttonProps
+}) => {
+  useFormikValuesChanged(() => {
+    !result.isIdle && clear();
+  });
+
+  return result.match({
+    Idle: () => (
+      <Button {...buttonProps}>
+        <Icon />
+        &nbsp;{label}
+      </Button>
+    ),
+    Loading: () => (
+      <Button {...buttonProps} disabled>
+        <Spinner as="span" animation="border" size="sm" role="loader" />
+        &nbsp;{label}
+      </Button>
+    ),
+    Rejected: (error) => (
+      <>
+        <Button {...buttonProps} variant="danger">
+          <BsExclamationCircle role="error" />
+          &nbsp;{label}
+        </Button>
+        <Form.Text className="text-danger d-inline-block mx-3">
+          {(error?.response?.data?.error || error.message).toString()}
```

Comment:
Czy jeśli `error` będzie `undefined`, to próba resolve'a do `error.message` rzuci błędem zamiast wyświetlić `error.message.toString`?

Reply:
tak jest, załatam